### PR TITLE
Dockerfile: some refactoring, and switch to "buster" variant

### PR DESCRIPTION
**Pull request by thaJeztah**
Debian buster is the current stable, so might be good to switch to that version; also did some refactoring of the dockerfile (see individual commits for details).

I'll open a follow-up PR to update to go 1.13, but thought to first make the switch to "buster" to keep changes smaller.

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
LGTM

**Comment by thaJeztah** [comment]
@cyphar @crosbymichael @kolyshkin LGTY?

File: Dockerfile [hunk 1]
```diff
@@ -9,32 +9,39 @@ ARG DEBIAN_FRONTEND
...
-    python-minimal \
-    uidmap \
-    kmod \
-    crossbuild-essential-armel crossbuild-essential-armhf crossbuild-essential-arm64 crossbuild-essential-ppc64el \
-    libseccomp-dev:armel libseccomp-dev:armhf libseccomp-dev:arm64 libseccomp-dev:ppc64el \
-    --no-install-recommends \
-    && apt-get clean
+    && dpkg --add-architecture ppc64el
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
```
Comment (kolyshkin) [inline Dockerfile]: What is the reason for splitting a single RUN statement into two?

Comment (thaJeztah) [reply Dockerfile]: The `dpkg --add-architecture` does not have to be re-run when we do `apt-get update`, and I thought it was slightly cleaner, but let me know if you feel strongly about that

Comment (kolyshkin) [reply Dockerfile]: I like a single `RUN` slightly better, but I'm fine either way

**Comment by AkihiroSuda** [comment]
needs rebase

**Comment by kolyshkin** [comment]
@thaJeztah could you please rebase it?

@mrunalp can we merge it? It is needed to unblock https://github.com/opencontainers/runc/pull/2239 which, in turn, is needed to unblock a few PRs that require golang 1.13

**Comment by mrunalp** [comment]
Yeah, but it needs a rebase now :/

**Comment by thaJeztah** [comment]
@kolyshkin @mrunalp @AkihiroSuda sorry for the delay; rebased and combined the two `RUN`'s again while at it

**Comment by AkihiroSuda** [comment]
LGTM

**Comment by kolyshkin** [comment]
CI failure is unrelated and is fixed by https://github.com/opencontainers/runc/pull/2268

**Comment by mrunalp** [comment]
LGTM

**Comment by crosbymichael** [comment]
can you rebase for the CI fix?

**Comment by thaJeztah** [comment]
Rebased 👍

**Comment by AkihiroSuda** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Comment by AkihiroSuda** [comment]
Flaky failure in systemd test seems unrelated. Merging.
```
$ travis_setup_go
gimme: given '1.13.x' but no release for '1.13' found
gimme: given '1.13.x' but no release for '1.13' found
Failed to run gimme
The command "travis_setup_go" failed and exited with 86 during .
```

https://travis-ci.org/github/opencontainers/runc/jobs/668927361?utm_medium=notification&utm_source=github_status

**Pull request merged**